front-end/src/domains/game-board/components/game-board/game-board.component.tsx

```diff
@@ -0,0 +1,195 @@
+import {GameHex} from "./game-hex.component";
+import React from "react";
+
+type Node = {
+  type: NodeType;
+}
+
+type NodeType = "water" | "space" | "concrete" | "fuel" | "food" | "player" | "metal" | "empty";
+const landTypes: NodeType[] = ["water", "concrete", "fuel", "food", "metal"];
+
+type Grid = Node[][];
+
+interface BoardProps {
+  savedBoard?: string;
+}
+export const Board: React.FC<BoardProps> = ({savedBoard}) => {
+  //If Saved Board is sent in load that, if not use placeholder board
+  const grid = (savedBoard)? JSON.parse(savedBoard) : placeholderBoard;
+
+  function createBoard(size: number, hexSize: number) {
```

Comment:
This is an awesome place to introduce memoization! So when a component re-renders, this function would need to be re-computed every time. But we can 'memoize' the function with the `useCallback` hook! Try this:
```suggestion
  const createBoard = useCallback((size: number, hexSize: number) => {
      const matrix = [];
    for (let x = 0; x <= size; x++) {
      const row = [];

      for (let y = 0; y <= size; y++) {
        if(grid[x][y].type != "empty" && grid[x][y].type != "space" && grid[x][y].type != "player") {
          const type = landTypes[Math.floor(Math.random() * landTypes.length)];
          row.push(<GameHex key={`${x}-${y}`} size={hexSize} xPos={x} yPos={y} type={type}/>);
        } else {
          row.push(<GameHex key={`${x}-${y}`} size={hexSize} xPos={x} yPos={y} type={grid[x][y].type}/>);
        }
      }
      matrix.push(row);
    }

    return matrix;
  }, [grid, landTypes])
```

Likewise, on line 18, the value of `grid` would need to recompute after every re-render. You can see how this would slow down / get expensive FAST when people are constantly changing state and the game board needs to refresh. Luckily, we can memoize values / variables as well. Give that a shot with this `useMemo` hook:

```
    const grid = useMemo(() => {
       return savedBoard ? JSON.parse(savedBoard) : placeholderBoard;
    }, [savedBoard, placeholderBoard]) 
```

You'll be able to import both of these hooks from `react` itself i.e.:
```
  import { useCallback, useMemo } from 'react';
```

Final thoughts: we won't have to do this for much longer lol - React 19 will come with auto-memoization out of the box. But for now it's good to know that these are things we need to be aware of, and how to optimize for UI that changes a lot.